Engdocs 1463 (#370)
compose-spec/compose-spec: 05-services.md

### domainname

> `domainname` declares a custom domain name to use for the service container. must be a valid RFC 1123 hostname.

Reviewer: ```suggestion
`domainname` declares a custom domain name to use for the service container. Must be a valid RFC 1123 hostname.
```

env_file: .env
```

> `env_file` can also be a list. The files in the list must be processed from the top down. For the same variable

Reviewer 2: ```suggestion
`env_file` can also be a list. The files in the list are processed from the top down. For the same variable
```

> specified in two env files, the value from the last file in the list must stand.

Reviewer 2: ```suggestion
specified in two env files, the value from the last file in the list is used.
```

```yml
env_file:
  - ./a.env
  - ./b.env
```

> Relative path must be resolved from the Compose file's parent folder. As absolute paths prevent the Compose

Reviewer 2: ```suggestion
Relative path are resolved from the Compose file's parent folder. As absolute paths prevent the Compose
```

    file.
  - Absolute path.

Service denoted by `service` must be present in the identified referenced Compose file.
Compose returns an error if:

- Service denoted by `service` was not found
- Compose file denoted by `file` was not found

#### Merging service definitions

Two service definitions (_main_ one in the current Compose file and _referenced_ one

> specified by `extends`) must be merged in the following way:

Reviewer 2: ```suggestion
specified by `extends`) are merged in the following way:
```

One exception that applies to `healthcheck` is that _main_ mapping cannot specify
`disable: true` unless _referenced_ mapping also specifies `disable: true`. Compose

> implementations must return an error in this case.

Reviewer 2: ```
Compose
implementations must return an error in this case.
```
missed this one due to line break
```suggestion
returns an error in this case.
```

  - "6060:6060/udp"
```

> **Note**: Host IP mapping may not be supported on the platform, in such case Compose rejects

> > the Compose file and must inform the user they will ignore the specified host IP.

Reviewer 2: ```suggestion
> the Compose file and informs the user it will ignore the specified host IP.
```

`profiles` defines a list of named profiles for the service to be enabled under. When not set, service is always enabled.

If present, `profiles` should follow the regex format of `[a-zA-Z0-9][a-zA-Z0-9_.-]+`.

### pull_policy

`pull_policy` defines the decisions Compose makes when it starts to pull images. Possible values are:

* `always`: Compose always pull the image from the registry.

> * `never`: Compose doesn't pull the image from a registry and should rely on the platform cached image.

Reviewer 2: ```suggestion
* `never`: Compose doesn't pull the image from a registry and relies on the platform cached image.
```

    external: true
```

Services may be granted access to multiple secrets. Long and short syntax for secrets may be used in the

> same Compose file. Defining a secret in the top-level `secrets` must not imply granting any service access to it.

Reviewer 2: ```suggestion
same Compose file. Defining a secret in the top-level `secrets` does not imply granting any service access to it.
```
